.github/workflows/nginx.yaml

```diff
@@ -61,45 +98,134 @@ jobs:
             ~/.cargo/registry/index/
             ~/.cargo/registry/cache/
             ~/.cargo/git/db/
-            nginx/objs/ngx_rust_examples
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: ${{ runner.os }}-cargo-
+            nginx/objs/**/CACHEDIR.TAG
+            nginx/objs/**/ngx-debug
+            nginx/objs/**/ngx-release
+          key: ${{ runner.os }}-nginx-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-nginx-
 
       - name: Update configure arguments
         if: matrix.nginx-ref != 'stable-1.24'
         run: |
-            echo NGX_CONFIGURE="${NGX_CONFIGURE} --with-http_v3_module" \
+            echo NGX_CONFIGURE_UNIX="${NGX_CONFIGURE_UNIX} --with-http_v3_module" \
                 >> "$GITHUB_ENV"
 
       - name: Configure nginx with static modules
         if: matrix.module == 'static'
         working-directory: nginx
         run: |
-          ${NGX_CONFIGURE} \
-              --add-module=${{ github.workspace }}/examples
+          ${NGX_CONFIGURE_CMD} \
+              ${NGX_CONFIGURE_UNIX} \
+              ${NGX_CONFIGURE_STATIC_MODULES}
 
       - name: Configure nginx with dynamic modules
         if: matrix.module != 'static'
         working-directory: nginx
-        env:
-          TEST_NGINX_GLOBALS: >-
-            load_module ${{ github.workspace }}/nginx/objs/ngx_http_async_module.so;
-            load_module ${{ github.workspace }}/nginx/objs/ngx_http_awssigv4_module.so;
-            load_module ${{ github.workspace }}/nginx/objs/ngx_http_curl_module.so;
-            load_module ${{ github.workspace }}/nginx/objs/ngx_http_upstream_custom_module.so;
         run: |
-          ${NGX_CONFIGURE} \
-              --add-dynamic-module=${{ github.workspace }}/examples
-          echo "TEST_NGINX_GLOBALS=$TEST_NGINX_GLOBALS" >> $GITHUB_ENV
+          ${NGX_CONFIGURE_CMD} \
+              ${NGX_CONFIGURE_UNIX} \
+              ${NGX_CONFIGURE_DYNAMIC_MODULES}
+          echo TEST_NGINX_GLOBALS="$NGX_TEST_GLOBALS_DYNAMIC" >> "$GITHUB_ENV"
 
       - name: Build nginx
         working-directory: nginx
         run: make -j$(nproc)
 
       - name: Run tests
         env:
+          PERL5LIB: ${{ github.workspace }}/nginx/tests/lib
           TEST_NGINX_BINARY: ${{ github.workspace }}/nginx/objs/nginx
           TEST_NGINX_MODULES: ${{ github.workspace }}/nginx/objs
+          TEST_NGINX_VERBOSE: 1
+        run: |
+          prove -j$(nproc) --state=save ${NGX_TEST_FILES} || prove -v --state=failed
+
+  windows:
+    runs-on: windows-2022
+    env:
+      TEMP: "C:\\TEMP"
+      TMP: "C:\\TEMP"
+      TMPDIR: "C:\\TEMP"
+      VCARCH: x64
+
+    strategy:
+      fail-fast: false
+      matrix:
+        nginx-ref:
+          - master
+        module:
+          - static
+
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
Not really related to the PR, but we should use full sha ids in actions instead of the mutable tags